src/main/java/uk/gov/hmcts/reform/coh/controller/EventForwardingController.java

```diff
@@ -116,19 +117,28 @@ public ResponseEntity resetSessionEvents(@Valid @RequestBody ResetSessionEventRe
             return ResponseEntity.status(HttpStatus.FAILED_DEPENDENCY).body("No registers for this jurisdiction & event type");
         }
 
-        Optional<SessionEventForwardingState> pendingEventForwardingState = sessionEventForwardingStateService.retrieveEventForwardingStateByName(SessionEventForwardingStates.EVENT_FORWARDING_PENDING.getStateName());
-        if(!pendingEventForwardingState.isPresent()) {
-            log.error("Pending event forwarding state was not found in the database");
+        try {
+            SessionEventForwardingState pendingEventForwardingState = getSessionEventForwardingState(SessionEventForwardingStates.EVENT_FORWARDING_PENDING);
+
+            SessionEventForwardingState failedEventForwardingState = getSessionEventForwardingState(SessionEventForwardingStates.EVENT_FORWARDING_FAILED);
+
+            sessionEventService.retrieveAllByEventForwardingRegister(sessionEventForwardingRegister.get()).stream()
```

Comment:
You can remove this filter by using Spring Repo to search for a specific forwarding state:
findAllBySessionEventForwardingRegisterAndSessionEventForwardingState